.github/workflows/community_ci.yml

```diff
@@ -62,68 +30,148 @@ jobs:
+        # This path refers to the fork .github folder.
+        # We make sure its content is in sync with the main repo .github folder by pulling it in the previous step
         uses: ./.github/actions/run-airbyte-ci
         with:
-          context: "manual"
-          dagger_cloud_token: ${{ secrets.DAGGER_CLOUD_TOKEN }}
+          # TODO remove this binary URL before merging
+          airbyte_ci_binary_url: https://storage.googleapis.com/dev-airbyte-cloud-connector-metadata-service/airbyte-ci/releases/ubuntu/3570293/airbyte-ci
+          context: "pull_request"
+          dagger_cloud_token: ${{ secrets.DAGGER_CLOUD_TOKEN_2 }}
           docker_hub_password: ${{ secrets.DOCKER_HUB_PASSWORD }}
           docker_hub_username: ${{ secrets.DOCKER_HUB_USERNAME }}
-          gcp_gsm_credentials: ${{ secrets.GCP_GSM_CREDENTIALS }}
+          git_repo_url: ${{ github.event.pull_request.head.repo.clone_url }}
+          gcs_credentials: ${{ secrets.METADATA_SERVICE_PROD_GCS_CREDENTIALS }}
           sentry_dsn: ${{ secrets.SENTRY_AIRBYTE_CI_DSN }}
-          git_branch: ${{ steps.extract_branch.outputs.branch }}
-          git_revision: ${{ steps.fetch_last_commit_id_pr.outputs.commit_id }}
-          github_token: ${{ env.PAT }}
-          s3_build_cache_access_key_id: ${{ secrets.SELF_RUNNER_AWS_ACCESS_KEY_ID }}
-          s3_build_cache_secret_key: ${{ secrets.SELF_RUNNER_AWS_SECRET_ACCESS_KEY }}
-          subcommand: "connectors ${{ github.event.inputs.test-connectors-options }} test"
-      - name: Test connectors [PULL REQUESTS]
-        if: github.event_name == 'pull_request_target'
+          github_token: ${{ github.token }}
+          subcommand: "format check all"
+          is_fork: "true"
+
+  connectors_test:
+    name: Run connectors tests on fork
+    needs: determine_environment
+    if: needs.determine_environment.outputs.environment == 'community-ci'
+    environment: ${{ needs.determine_environment.outputs.environment }}
+    runs-on: community-connector-test-large
+    timeout-minutes: 180 # 3 hours
+    permissions:
+      statuses: write
+    env:
+      GITHUB_HEAD_REF: ${{ github.head_ref }}
+      MAIN_BRANCH_NAME: "augustin/04-18-community-ci_workflow_prevent_injection"
+      # TODO
+      #MAIN_BRANCH_NAME: "master"
+
+    steps:
+      - name: Checkout fork
+        uses: actions/checkout@v4
+        with:
+          # This checkouts a fork which can contain untrusted code
+          # It's deemed safe as the community-ci environment requires manual reviewer approval to run
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          ref: ${{ github.head_ref }}
+          fetch-depth: 1
+
+      # On checkout the head commit is a detached head
+      # We have to use the git CLI to get the last commit id of the head branch
+      # We also perform validation and sanitization on the GITHUB_HEAD_REF to prevent command injection
+      - name: Fetch last commit id from remote branch
```

Comment:
These two steps are used in both commands (format and test) — should we ref them out? Is that a thing in gh actions at all, without making a custom action? ;)

Reply:
Yes we can avoid code duplication here. I did not do it yet as I wanted to make sure the full flow works first (#draft-mode) .